### ISSUE TITLE
[TypeError: Cannot read property 'Kerberos' of undefined] on server start

### ISSUE THREAD

**Issue by peterbsmyth** [peterbsmith2/api-a-day#1]
[TypeError: Cannot read property 'Kerberos' of undefined]

**Comment by peterbsmyth** [comment]
i have no idea what's generating this error. it doesn't seem to impact the API.

**Comment by laggingreflex** [comment]
I'm not sure what this library is but I too just started getting this exact same error, and google landed me here. I just updated all my node_modules to latest versions. I'm guessing one of them has a breaking change or something. I'm using Koa app, and going through your package.jsons I think mongoose seems to be common in both our apps.

**Comment by davegri** [comment]
Getting this error using a very simple node app, also using mongoose, got here from google (not using this library)

**Comment by ghost** [comment]
> @davegri I've also just noticed this in my console, I've just started a new project and literally have nothing but a basic web server. `TypeError: Cannot read property 'Kerberos' of undefined`

**Edit:**
@davegri @laggingreflex @peterbsmith2 I've just disabled Mongoose in my project and i'm no longer seeing the message, as you suggested mongoose seems to be the culprit.

**Comment by Colure** [comment]
@davegri @laggingreflex @peterbsmith2 I've just disabled Mongoose in my project and i'm no longer seeing the message, as you suggested mongoose seems to be the culprit.

**Comment by davegri** [comment]
@Colure, yeah it's definitely mongoose, it's related to the mongodb drivers, seems harmless

**Comment by laggingreflex** [comment]
This is where the error is coming from:

```
…\mongodb-core\lib\auth\sspi.js:
   25: // Try to grab the Kerberos class
   26  try {
   27:   Kerberos = require_optional('kerberos').Kerberos
   28    // Authentication process for Mongo
   29:   MongoAuthProcess = require_optional('kerberos').processes.MongoAuthProcess
   30  } catch(err) {
   31    console.dir(err)
```

**Comment by ghost** [comment]
@davegri @laggingreflex Perhaps it's not as harmless as you think, I myself have very little knowledge on the subject of Kerbos.

> Kerberos is a secure method for authenticating a request for a service in a computer network. Kerberos was developed in the Athena Project at the Massachusetts Institute of Technology (MIT).

However, just based on the fact it's regarding authentication, It might be worth while rolling back a version on production applications, just incase.

**Comment by nedyalkov** [comment]
The error doesn't exist with mongoose ~4.3.0 So I guess it is some problem with the last release (from yesterday).

**Comment by peterbsmyth** [comment]
I changed my mongoose version from 4.4.2 to 4.4.1 and that removed the error.  

Apparently mongoose 4.4.1 depends on mongodb 2.1.4 which depends on kerberos 0.0.18.  

mongoose 4.4.2 depends on mongodb 2.1.5 which removes kerberos as a dependency.  

So likely mongoose was expecting Kerberos to be there when it no longer is.

**Comment by peterbsmyth** [comment]
This issue is documented at Automattic/mongoose#3860

**Comment by peterbsmyth** [comment]
fixed in Automattic/mongoose#3861

**Comment by mirabel-koso** [comment]
I just restarted my system and the error went away :)
